src/mlio-py/mlio/core/data_reader.cxx

```diff
@@ -280,46 +273,11 @@ register_data_readers(py::module &m)
              })
         .def("__next__", &py_data_iterator::next);
 
-    py::class_<data_reader, py_data_reader, intrusive_ptr<data_reader>>(
+    py::class_<data_reader_params, std::shared_ptr<data_reader_params>>(
```

Comment:
`py::class_<data_reader_params>` should be enough. Is there a specific reason why you picked `std::shared_ptr` instead of `std::unique_ptr` (the default) here?

Reply:
The [pybind11 docs](https://pybind11.readthedocs.io/en/stable/advanced/smart_ptrs.html) say it is illegal to pass a `std::unique_ptr` as a function argument. I originally tried with unique pointers, and was getting errors related to the unique pointers during build.